corehq/sql_db/routers.py

```diff
@@ -43,6 +44,8 @@ def allow_migrate(db, app_label):
         )
     elif app_label == SQL_ACCESSORS_APP:
         return db in partition_config.get_form_processing_dbs()
+    elif app_label == WAREHOUSE_APP:
```

Comment:
i think we'll need to add `hasattr(settings, "WAREHOUSE_DATABASE_ALIAS")` to this condition as well, otherwise if that setting is not present then it would always return False on the next line and the migration wouldn't run in any db